packages/fork-choice/test/unit/protoArray/executionStatusUpdates.test.ts

```diff
@@ -0,0 +1,645 @@
+ *
+ *  0 (PreMerge) <- 1A (Syncing) <- 2A (Syncing) <- 3A (Syncing)
+ *                               ^- 2B (Syncing) <- 3B (Syncing)
+ *                               ~~ 2C (Syncing) <- 3C (Syncing)
+ */
+
+function setupForkChoice(): ProtoArray {
+  const fc = ProtoArray.initialize(
+    {
+      slot: 0,
+      stateRoot: "-",
+      parentRoot: "-",
+      blockRoot: "0",
+
+      justifiedEpoch: 0,
+      justifiedRoot: "-",
+      finalizedEpoch: 0,
+      finalizedRoot: "-",
+
+      ...{executionPayloadBlockHash: null, executionStatus: ExecutionStatus.PreMerge},
+    } as Omit<ProtoBlock, "targetRoot">,
+    0
+  );
+
+  for (const block of blocks) {
+    const executionData = (block.executionStatus === ExecutionStatus.PreMerge
+      ? {executionPayloadBlockHash: null, executionStatus: ExecutionStatus.PreMerge}
+      : {executionPayloadBlockHash: block.root, executionStatus: block.executionStatus}) as BlockExecution;
+    fc.onBlock(
+      {
+        slot: block.slot,
+        blockRoot: block.root,
+        parentRoot: block.parent,
+        stateRoot: "-",
+        targetRoot: "-",
+
+        justifiedEpoch: 0,
+        justifiedRoot: "-",
+        finalizedEpoch: 0,
+        finalizedRoot: "-",
+
+        unrealizedJustifiedEpoch: 0,
+        unrealizedJustifiedRoot: "-",
+        unrealizedFinalizedEpoch: 0,
+        unrealizedFinalizedRoot: "-",
+
+        ...executionData,
+      },
+      block.slot
+    );
+  }
+
+  const deltas = Array.from({length: fc.nodes.length}, () => 0);
+  fc.applyScoreChanges({
+    deltas,
+    proposerBoost: null,
+    justifiedEpoch: 0,
+    justifiedRoot: "-",
+    finalizedEpoch: 0,
+    finalizedRoot: "-",
+    currentSlot: 3,
+  });
+  return fc;
+}
+
+describe("executionStatus / normal updates", () => {
+  const fc = setupForkChoice();
+
+  /**
+   *  0 (PreMerge) <- 1A (Syncing) <- 2A (Syncing) <- 3A (Syncing)
+   *                               ^- 2B (Syncing) <- 3B (Syncing)
+   *                               ~~ 2C (Syncing) <- 3C (Syncing)
+   */
+  const preValidation = collectProtoarrayValidationStatus(fc);
+  it("preValidation forkchoice setup should be correct", () => {
+    expect(preValidation).to.be.deep.equal(expectedPreValidationFC);
+  });
+
+  /**
+   * Invalidate 3C with LVH on 2C which stays in Syncing
+   *
+   *  0 (PreMerge) <- 1A (Syncing) <- 2A (Syncing) <- 3A (Syncing)
+   *                               ^- 2B (Syncing) <- 3B (Syncing)
+   *                               ~~ 2C (Syncing) <- 3C (Invalid)
+   */
+  fc.validateLatestHash(
+    {
+      executionStatus: ExecutionStatus.Invalid,
+      latestValidExecHash: "2C",
+      invalidateFromBlockHash: "3C",
+    },
+    3
+  );
+
+  const invalidate3CValidate2CForkChoice = collectProtoarrayValidationStatus(fc);
+  it("correcly invalidate 3C and validate 2C only", () => {
+    expect(invalidate3CValidate2CForkChoice).to.be.deep.equal([
```

Comment:
This assert state can be declared in one line? Or are longer than 120 characters?

Reply:
yea missed updating this one

Reply:
updated